dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs

```diff
@@ -27,9 +27,9 @@ internal partial class ClientCore
     internal static string DeploymentNameKey => "DeploymentName";
 
     /// <summary>
-    /// Model Id or Deployment Name
+    /// Deployment name or model Id.
     /// </summary>
-    internal string DeploymentOrModelName { get; set; } = string.Empty;
+    internal string DeploymentNameOrModelId { get; set; } = string.Empty;
```

Comment:
We should have two seperate properties as these two things are not related e.g. I could have a deploy name as `eastus-gpt-4o` and model id as `gpt-4o`.

Reply:
The property was renamed to `DeploymentName`. Right now, the Azure.AI.OpenAI SDK doesn't let you specify both at the same time and only works with the deployment name. So, there's no need for two properties; the deployment name should be enough for now. 

There are also places in SK where the deployment name is mapped/assigned to the model id. This seems confusing and wrong. This issue captures the problem and will be used to fix it: https://github.com/microsoft/semantic-kernel/issues/7142